pkg/config/config.go

```diff
@@ -223,23 +244,33 @@ func (gc *Config) addKubernetesServiceHostInNoProxy() {
 	}
 }
 
-func (gc *Config) addSecretDataToConfig(secret *v1.Secret) {
+func (gc *Config) addSecretDataToConfig(secret *v1.Secret) error {
 	extractedValues := map[string]string{}
 	for key, value := range secret.Data {
 		extractedValues[key] = string(value)
 	}
-
-	gc.addDataToConfig(extractedValues)
+	return gc.addDataToConfig(extractedValues)
 }
 
-func (gc *Config) addConfigMapDataToConfig(configMap *v1.ConfigMap) {
-	gc.addDataToConfig(configMap.Data)
+func (gc *Config) addConfigMapDataToConfig(configMap *v1.ConfigMap) error {
+	return gc.addDataToConfig(configMap.Data)
 }
 
-func (gc *Config) addDataToConfig(data map[string]string) {
+func (gc *Config) addDataToConfig(data map[string]string) error {
 	gc.caCerts = data[caCertsKey]
 	gc.httpProxy = data[httpProxyKey]
 	gc.httpsProxy = data[httpsProxyKey]
 	gc.noProxy = data[noProxyKey]
 	gc.skipTLSVerify = data[skipTLSVerifyKey]
+
+	if val := data[kappDeployRawOptionsKey]; len(val) > 0 {
+		var opts []string
+		err := json.Unmarshal([]byte(val), &opts)
+		if err != nil {
+			return fmt.Errorf("Unmarshaling kappDeployRawOptions as JSON: %s", err)
+		}
+		gc.kappDeployRawOptions = opts
```

Comment:
i just dug around to remind myself where the allow-list is for this feature,  since at a glance it looks like we're taking unverified user input and passing on the command line.   how would you feel about adding a comment that says something like,  `// don't worry, input is verified before kapp is invoked. see pkg/deploy/kapp_restrict.go` ?

Reply:
fixed.